kolibri/content/models.py

```diff
@@ -144,76 +123,50 @@ def get_descendant_content_ids(self):
-        descendants = ContentNode.objects.filter(lft__gte=self.lft, lft__lte=self.rght).exclude(kind="'{}'".format(content_kinds.TOPIC))
-        # extract the unique pairs of content_id and kind, as a queryset
-        unique_content_id_kinds = descendants.values("content_id", "kind").order_by("content_id", "kind").distinct().values("kind")
-        # construct and execute a SQL query to count the number of nodes with unique content_ids for each kind
-        query = 'SELECT "kind", COUNT(kind) as count FROM ({}) GROUP BY kind'.format(str(unique_content_id_kinds.query))
-        conn = get_content_database_connection(self._state.db)
-        # turn the results into a dict, mapping kind into unique count
-        return dict(conn.execute(query))
-
 
 @python_2_unicode_compatible
-class Language(ContentDatabaseModel):
+class Language(models.Model):
     id = models.CharField(max_length=14, primary_key=True)
     lang_code = models.CharField(max_length=3, db_index=True)
     lang_subcode = models.CharField(max_length=10, db_index=True, blank=True, null=True)
     # Localized name
     lang_name = models.CharField(max_length=100, blank=True, null=True)
-    lang_direction = models.CharField(max_length=3, choices=LANGUAGE_DIRECTIONS)
-
-    objects = ContentQuerySet.as_manager()
+    lang_direction = models.CharField(max_length=3, choices=LANGUAGE_DIRECTIONS, default=LANGUAGE_DIRECTIONS[0][0])
 
     def __str__(self):
         return self.lang_name
 
 
-@python_2_unicode_compatible
-class File(ContentDatabaseModel):
+class File(models.Model):
     """
-    The bottom layer of the contentDB schema, defines the basic building brick for content.
+    The second to bottom layer of the contentDB schema, defines the basic building brick for content.
     Things it can represent are, for example, mp4, avi, mov, html, css, jpeg, pdf, mp3...
     """
     id = UUIDField(primary_key=True)
-    checksum = models.CharField(max_length=400, blank=True)
-    extension = models.CharField(max_length=40, choices=file_formats.choices, blank=True)
+    # The foreign key mapping happens here as many File objects can map onto a single local file
+    local_file = models.ForeignKey('LocalFile', related_name='files')
     available = models.BooleanField(default=False)
-    file_size = models.IntegerField(blank=True, null=True)
-    contentnode = models.ForeignKey(ContentNode, related_name='files', blank=True, null=True)
+    contentnode = models.ForeignKey(ContentNode, related_name='files')
     preset = models.CharField(max_length=150, choices=format_presets.choices, blank=True)
     lang = models.ForeignKey(Language, blank=True, null=True)
     supplementary = models.BooleanField(default=False)
     thumbnail = models.BooleanField(default=False)
-    priority = models.IntegerField(blank=True, null=True)
-
-    objects = ContentQuerySet.as_manager()
+    priority = models.IntegerField(blank=True, null=True, db_index=True)
```

Comment:
Not sure, added it just because it is the default ordering, and so affects all queries.